Code/GraphMol/RGroupDecomposition/RGroupDecompData.h

```diff
@@ -20,7 +21,7 @@ namespace RDKit
 {
 struct RGroupDecompData {
   // matches[mol_idx] == vector of potential matches
-  std::map<int, RWMol> cores;
+  std::map<int, RCore> cores;
```

Comment:
could this be a vector, instead? It looks like it's always contiguous. Then you'd be able to remove so much `.second`

Reply:
Not without major refactoring, as `core_idx` can be negative, as I learnt trying your suggestion.

Reply:
The explicitly aren’t consecutive as some may come from users as well,  R1...R3...R7,  pesky users :). Also, the negative versus positive indicate which are user defined and which need to be added.